### DISCUSSION TITLE
Changing the input graph for calcite dissolution simulation tutorial.

### DISCUSSION THREAD

**Discussion by YuxuanZZ** [GeoChemFoam/GeoChemFoam#26] [Q&A] [answered]
Hi,

First of all, thank you so much for helping me get a better understanding of reaction multiphase flow.

I would like to use the solver, reactiveTransportDBSFoam, to do a simple dissolution of a calcite dominated pore structure in an acidic environment! I found that replacing the image of one of the tutorials( Dissolution of a 3D calcite post with DBS) would complete the simulation.

I cropped the Bentheimer_2d25um_binary.raw( https://www.digitalrocksportal.org/projects/317/origin_data/1360/) to 144*144*144, and set it as the input image of DBS calcite dissolution tutorial case. 

After running reactiveTransportDBSFoam for small time to refine mesh at interface, I found error below in 'reactiveTransportDBSFoam.out':

Selected 30083 cells for refinement out of 46656.
Refined from 46656 to 257237 cells.
--> FOAM Warning : 
    From virtual void Foam::dynamicRefineFvMesh::mapFields(const Foam::mapPolyMesh&)
    in file dynamicRefineFvMesh/dynamicRefineFvMesh.C at line 312
    Cannot find surfaceScalarField nEpsf in user-provided flux mapping table 
1
(
phi U
)
    The flux mapping table is used to recreate the flux on newly created faces.
    Either add the entry if it is a flux or use (nEpsf none) to suppress this warning.
Selected 0 split points out of a possible 30083.

Also, I found out that my 'C' won't converge while 'p' and 'U' can converge.

Best regards,
Yuxuan Zhang

**Comment by GeoChemFoam** [comment]
Hi,

Have you checked the initial mesh with paraview? Does it have the correct boundary conditions and inlet/outlet?


Best wishes,

Hannah

From: YuxuanZZ ***@***.***>
Sent: Monday, October 16, 2023 5:10 PM
To: GeoChemFoam/GeoChemFoam ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [GeoChemFoam/GeoChemFoam] Changing the input graph for calcite dissolution simulation tutorial. (Discussion #26)

****************************************************************
Caution: This email originated from a sender outside Heriot-Watt University.
Do not follow links or open attachments if you doubt the authenticity of the sender or the content.
****************************************************************


Hi,

First of all, thank you so much for helping me get a better understanding of reaction multiphase flow.

I would like to use the solver, reactiveTransportDBSFoam, to do a simple dissolution of a calcite dominated pore structure in an acidic environment! I found that replacing the image of one of the tutorials( Dissolution of a 3D calcite post with DBS) would complete the simulation.

I cropped the Bentheimer_2d25um_binary.raw( https://www.digitalrocksportal.org/projects/317/origin_data/1360/) to 144144144, and set it as the input image of DBS calcite dissolution tutorial case.

After running reactiveTransportDBSFoam for small time to refine mesh at interface, I found error below in 'reactiveTransportDBSFoam.out':

Selected 30083 cells for refinement out of 46656.
Refined from 46656 to 257237 cells.
--> FOAM Warning :
From virtual void Foam::dynamicRefineFvMesh::mapFields(const Foam::mapPolyMesh&)
in file dynamicRefineFvMesh/dynamicRefineFvMesh.C at line 312
Cannot find surfaceScalarField nEpsf in user-provided flux mapping table
1
(
phi U
)
The flux mapping table is used to recreate the flux on newly created faces.
Either add the entry if it is a flux or use (nEpsf none) to suppress this warning.
Selected 0 split points out of a possible 30083.

Also, I found out that my 'C' won't converge while 'p' and 'U' can converge.

Best regards,
Yuxuan Zhang

—
Reply to this email directly, view it on GitHub<https://github.com/GeoChemFoam/GeoChemFoam/discussions/26>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AWDOAFXU4AD3LI6DHKTBEWDX7VL4NAVCNFSM6AAAAAA6COHOZ6VHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZVG42DEOBTGY>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

________________________________

Founded in 1821, Heriot-Watt is a leader in ideas and solutions. With campuses and students across the entire globe we span the world, delivering innovation and educational excellence in business, engineering, design and the physical, social and life sciences. This email is generated from the Heriot-Watt University Group, which includes:

  1.  Heriot-Watt University, a Scottish charity registered under number SC000278
  2.  Heriot- Watt Services Limited (Oriam), Scotland's national performance centre for sport. Heriot-Watt Services Limited is a private limited company registered is Scotland with registered number SC271030 and registered office at Research & Enterprise Services Heriot-Watt University, Riccarton, Edinburgh, EH14 4AS.

The contents (including any attachments) are confidential. If you are not the intended recipient of this e-mail, any disclosure, copying, distribution or use of its contents is strictly prohibited, and you should please notify the sender immediately and then delete it (including any attachments) from your system.

**Reply by YuxuanZZ** [reply]
Dear Hannah,

Thank you for your replying.

The following graphs are my inlet/outlet and walls conditions respectively after running "./createMesh.sh".
![inlet outlet](https://github.com/GeoChemFoam/GeoChemFoam/assets/146457240/466168e3-c09a-4016-8ce8-ca680191dbef)
![walls](https://github.com/GeoChemFoam/GeoChemFoam/assets/146457240/00f25734-c90e-43ff-b0eb-fdd8b4b51ac4)
 
Also I attached the initial mesh of this graph by applying SnappyHexMesh for comparation.
![snappyhex](https://github.com/GeoChemFoam/GeoChemFoam/assets/146457240/51ac4bbf-4ba4-4d1d-875b-763a3410b642)

I ran this topography in two calcite dissolution cases. In the case applying ALE, my 'U' and 'p' can't converge after running for a small time step( 1e-6). In the case applying VOS methods, my 'C' can't converge in the first small time step( 1e-6).

Best regards,
Yuxuan Zhang

**Reply by julienmaes** [reply]
Hi Yuxuan,

It looks like your problem is with AMR. See if you can make it work without AMR first. Put nlevel=0 in createMesh and a use a cartesian mesh that fits exactly the dimension of the image

**Reply by YuxuanZZ** [reply]
Hi,

Thank you for your replying.

I've tried to set the nlevel=0 cartesian mesh fits exactly the dimension of the image.
![noAMR](https://github.com/GeoChemFoam/GeoChemFoam/assets/146457240/3225cc81-b4df-4c84-97a8-14e61253eb98)

However, after running 'runCase0.sh', terminal reported the same error as following:

run reactiveTransportDBSFoam for 1e-06
cp: cannot stat '1e-06/C': No such file or directory
cp: cannot stat '1e-06/U': No such file or directory
cp: cannot stat '1e-06/p': No such file or directory
cp: cannot stat '1e-06/phi': No such file or directory
cp: cannot stat '1e-06/R': No such file or directory
Note: Please check the last line of reactiveTransportDBSFoam0.out to confirm the equation have converged. If it has not, re-run script and/or change the tolerance and residual controls in system/fvSolution

I noticed that there is no '1e-06/*' may because 'C' and 'p' can't converge in this small time step. The residuals of 'C' and 'p' have been oscillating in 'reactiveTransportDBSFoam0.out'.
[reactiveTransportDBSFoam0.out.log](https://github.com/GeoChemFoam/GeoChemFoam/files/13063845/reactiveTransportDBSFoam0.out.log)

Thank you very much for your patience in answering my questions.

Best regards,
Yuxuan

**Comment by YuxuanZZ** [comment]
Hi,

I tried to increase the diffusion coefficient and the tolerance of 'C'.

My 'C' and 'p' will converged after '1e-06'( after running 'runCase0.sh'). But my residual of 'p' hardly decrease after running 'runCase.sh'.

I looked into the initial condition by paraview. I noticed there is a pressure increase at the rock matrix( eps=0) on inlet boundary.

![inlet_p](https://github.com/GeoChemFoam/GeoChemFoam/assets/146457240/4b3c5c3b-2c40-46fe-9f81-0f96ac404f0a)
(p on the inlet boundary)

![inlet_eps](https://github.com/GeoChemFoam/GeoChemFoam/assets/146457240/86a3f19c-01d0-4295-b3b9-a8e1a5b0f13e)
(eps of inlet boundary, eps=0 for rock and eps=1 for fluid)

I'm new to CFD, this probably is a silly idea. I'm wondering if it could be because the fluid is injected from the entire inlet boundary, so there is fluid deceleration on the rock matrix causing the pressure to rise. If the inlet boundary condition is made to be injected only from the pore matrix will it converge better?

Best regards,
Yuxuan

**Comment by YuxuanZZ** [comment, marked as answer]
Hi,

Hope you doing well.

After going through your related papers on GitHub and troubleshooting problem by problem, I realized that I didn't leave padding width at the entrance as well as exit boundaries, which caused the dissolution on boundaries.

Thank you,
Yuxuan

**Comment by GeoChemFoam** [comment]
Hi Yuxuan,

Great to hear you solved the issue.

Cheers,

Hannah

From: YuxuanZZ ***@***.***>
Date: Thursday, 2 November 2023 at 14:14
To: GeoChemFoam/GeoChemFoam ***@***.***>
Cc: Menke, Hannah ***@***.***>, Comment ***@***.***>
Subject: Re: [GeoChemFoam/GeoChemFoam] Changing the input graph for calcite dissolution simulation tutorial. (Discussion #26)
****************************************************************
Caution: This email originated from a sender outside Heriot-Watt University.
Do not follow links or open attachments if you doubt the authenticity of the sender or the content.
****************************************************************


Hi,

Hope you doing well.

After going through your related papers on GitHub and troubleshooting problem by problem, I realized that I didn't leave padding width at the entrance as well as exit boundaries, which caused the dissolution on boundaries.

Thank you,
Yuxuan

—
Reply to this email directly, view it on GitHub<https://github.com/GeoChemFoam/GeoChemFoam/discussions/26#discussioncomment-7457388>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AWDOAFWN6OFPBLWKGZLQ2VDYCOTEXAVCNFSM6AAAAAA6COHOZ6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TINJXGM4DQ>.
You are receiving this because you commented.Message ID: ***@***.***>

________________________________

Founded in 1821, Heriot-Watt is a leader in ideas and solutions. With campuses and students across the entire globe we span the world, delivering innovation and educational excellence in business, engineering, design and the physical, social and life sciences. This email is generated from the Heriot-Watt University Group, which includes:

  1.  Heriot-Watt University, a Scottish charity registered under number SC000278
  2.  Heriot- Watt Services Limited (Oriam), Scotland's national performance centre for sport. Heriot-Watt Services Limited is a private limited company registered is Scotland with registered number SC271030 and registered office at Research & Enterprise Services Heriot-Watt University, Riccarton, Edinburgh, EH14 4AS.

The contents (including any attachments) are confidential. If you are not the intended recipient of this e-mail, any disclosure, copying, distribution or use of its contents is strictly prohibited, and you should please notify the sender immediately and then delete it (including any attachments) from your system.
